docs/stories/WebTools.md

```diff
@@ -0,0 +1,53 @@
+# Web Tools
+
+## Configure web application directory in WTP configurations for web application projects
+
+### Motivation
+The `ide-integration.md` document in Gradle core specifies a story ["Expose web application directory"](...). The goal is to define an initial end-to-end framework for Buidship/WTP integration, and to
+communicate the location of the web application directory to WTP.
+
+### Estimate
+
+- ? day(s)
+
+### Implementation
+- Create branch in buildship repository `wtp-integration`.
+- Define `org.eclipse.buildship.wtp` plugin in `wtp-integration`.
+- Define `org.eclipse.buildship.wtp.test` plugin in `wtp-integration`.
+- Wait for corresponding story in TAPI to be completed.
+- Upgrade Tooling Commons (if necessary).
+- Define configuration class in `org.eclipse.buildship.wtp`, either configurator class, or a similar class to `DefaultWorkspaceGradleOperations`.
+    - Retrieve Web application model from TAPI
+    - If a project is a web application project (model will be `null` if project doesn't apply `war` plugin)
+        - Add Java and Dynamic Web Facets to project, which will generate component file
+        - Configure web app dir location in component file
+    - Otherwise, do nothing
+- Invoke WTP configuration from Buildship core
+
+### Test cases
+- Configuration only invoked on web application projects
+- Java facet is added to project
+- Dynamic Web Facet is added to project
+- If Java Facet already on project but has incorrect version, version is updated to correct version based on source Java version
+- If Dynamic Web Facet already on project but has incorrect version, version is updated to correct version
+- Default Java Facet version is x.
+- Default Dynamic Web Facet version is x.
+- If Java Facet exists on project with correct version, Facet is not affected.
+- If Dynamic Web Facet exists on project with correct version, Facet is not affected.
+- Custom web application directory location is communicated successfully to component file, file exists.
+- Default web application directory location is communicated successfully to component file, file exists.
+- Custom web application directory location is communicated successfuly to component file, file does not exist.
+- Default web application directory location is communicated successfully to component file, file does not exist.
+
```

Comment:
I'm fine if you implement it. Results will certainly be more correct than
if you don't. And I might steal your code to do something similar to
m2e-wtp when I find the time.

On Thu, Oct 22, 2015 at 3:52 PM, Ian Stewart-Binks <notifications@github.com
> wrote:

> In docs/stories/WebTools.md
> <https://github.com/eclipse/buildship/pull/181#discussion_r42797215>:
>
> > +
> > +### Test cases
> > +- Configuration only invoked on web application projects
> > +- Java facet is added to project
> > +- Dynamic Web Facet is added to project
> > +- If Java Facet already on project but has incorrect version, version is updated to correct version based on source Java version
> > +- If Dynamic Web Facet already on project but has incorrect version, version is updated to correct version
> > +- Default Java Facet version is x.
> > +- Default Dynamic Web Facet version is x.
> > +- If Java Facet exists on project with correct version, Facet is not affected.
> > +- If Dynamic Web Facet exists on project with correct version, Facet is not affected.
> > +- Custom web application directory location is communicated successfully to component file, file exists.
> > +- Default web application directory location is communicated successfully to component file, file exists.
> > +- Custom web application directory location is communicated successfuly to component file, file does not exist.
> > +- Default web application directory location is communicated successfully to component file, file does not exist.
> > +
>
> Alright, cool. That's what I suspected. Thanks.
>
> If it's ugly, but it's the only way to implement the functionality we
> want, then I don't see any reason not to do it. The question is whether or
> not we want that functionality.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/buildship/pull/181/files#r42797215>.
>



-- 
"Have you tried turning it off and on again" - The IT Crowd
And if that fails, then http://goo.gl/tnBgH5